MSAL/src/MSALPublicClientApplication.m

```diff
@@ -1434,12 +1435,74 @@ - (void)signoutWithAccount:(nonnull MSALAccount *)account
         block(NO, localError, nil);
         return;
     }
+
+    if (signoutParameters.wipeCacheForAllAccounts)
+    {
+        BOOL result = YES;
+        NSError *localError;
+        
+        result = [self.tokenCache clearCacheForAllAccountsWithContext:nil error:&localError];
+        
+        if (!result)
+        {
+            block(NO, localError, nil);
+            return;
+        }
+        
+#if !TARGET_OS_IPHONE
+        // Clear additional cache locations
+        NSDictionary<NSString *, NSDictionary *> *additionalPartnerLocations = MSALWipeCacheForAllAccountsConfig.additionalPartnerLocations;
+        if (additionalPartnerLocations && additionalPartnerLocations.count > 0)
+        {
+            NSMutableArray <NSString *> *locationErrors = nil;
+            MSIDMacACLKeychainAccessor *keychainAccessor = [[MSIDMacACLKeychainAccessor alloc] initWithTrustedApplications:nil accessLabel:@"Microsoft Credentials" error:nil];
+            for (NSString* locationName in additionalPartnerLocations)
+            {
+                NSDictionary *cacheLocation = additionalPartnerLocations[locationName];
+                
+                // Try to read the keychain data in order to trigger the prompt asking for login password, user HAS TO click 'Always Allow' to then be able to delete it.
+                [keychainAccessor getDataWithAttributes:cacheLocation
+                                                context:nil
+                                                  error:&localError];
+                
+                if (localError)
+                {
+                    NSError *additionalLocationError = MSIDCreateError(MSIDErrorDomain, MSIDErrorInternal, [NSString stringWithFormat:@"WipeCacheForAllAccounts - error when reading cache for the item: %@.", locationName], nil, nil, localError, nil, nil, YES);
```

Comment:
Should you proceed to the next one if one fails for best effort cleanup?

Reply:
yes, this was updated